test/modules/DirectMethodSender/DirectMethodSenderBase.cs

```diff
@@ -79,12 +79,14 @@ abstract class DirectMethodSenderBase : IDisposable
             {
                 try
                 {
+                    logger.LogDebug($"InvokeDirectMethodWithRetryAsync with count {directMethodCount}; Retry {transientRetryCount}");
                     transientRetryCount++;
                     resultStatus = await this.InvokeDeviceMethodAsync(deviceId, targetModuleId, methodName, directMethodCount, CancellationToken.None);
+                    return resultStatus;
```

Comment:
shouldn't it return on line 94?

Reply:
It would but that's not the expected behavior. 
if the DM can successfully be send, it should only send 1 dm, not 3 dms.

Reply:
I prefer not using return inside a while loop, would you use break statement or have a proper exit condition for while clause? or you still prefer use return here?

Reply:
break; sounds good. Fixed!